base/src/java/io/FileDescriptor.d

```diff
@@ -0,0 +1,184 @@
+module java.io.FileDescriptor;
+
+version(Posix) {
+    import core.stdc.errno : errno;
+    import core.stdc.string : strerror;
+    import core.sys.posix.unistd : fsync;
+} else version(Windows) {
+    import core.sys.windows.core;
+}
+
+import std.string : fromStringz;
+
+import java.io.SyncFailedException;
+
+/**
+ * Instances of the file descriptor class serve as an opaque handle to the
+ * underlying machine-specific structure representing an open file, an open
+ * socket, or another source or sink of bytes.
+ *
+ * The main practical use for a file descriptor is to create a
+ * `FileInputStream` or `FileOutputStream` to contain it.
+ *
+ * Applications should not create their own file descriptors.
+ *
+ * See_Also: FileInputStream, FileOutputStream
+ */
+public final class FileDescriptor
+{
+    private int fFD;
+    private void* fHandle;
+
+    /// UNIX file descriptor
+    package int getFD()
+    {
+        return fFD;
+    }
+
+    /// Windows file HANDLE
+    package void* getHandle()
+    {
+        return fHandle;
+    }
+
+    /**
+     * A handle to the standard input stream.
+     *
+     * Usually, this file descriptor is not used directly, but rather via the
+     * output stream known as `System.in`.
+     */
+    public static const FileDescriptor in_;
```

Comment:
I don't know if there are other static variables which are objects in the code base, but these are thread local storage, do we want that?